Sources/SwiftMemcache/MemcachedConnection.swift

```
@@ -0,0 +1,198 @@
+            bufferAllocator: ByteBufferAllocator,
+            /// The underlying channel to communicate with the server.
+            channel: NIOAsyncChannel<MemcachedResponse, MemcachedRequest>,
+            /// The stream of requests to be sent to the server.
+            requestStream: AsyncStream<StreamElement>,
+            /// The continuation for the request stream.
+            requestContinuation: AsyncStream<StreamElement>.Continuation
+        )
+        case finished
+    }
+
+    /// Enum representing the possible errors that can be encountered in `MemcachedConnection`.
+    enum MemcachedConnectionError: Error {
+        /// Indicates that the connection has shut down.
+        case connectionShutdown
+    }
+
+    private var connectionState: State
+
+    /// Initialize a new MemcachedConnection.
+    ///
+    /// - Parameters:
+    ///   - host: The host address of the Memcache server.
+    ///   - port: The port number of the Memcache server.
+    ///   - eventLoopGroup: The event loop group to use for this connection.
+    public init(host: String, port: Int, eventLoopGroup: EventLoopGroup) {
+        self.host = host
+        self.port = port
+        let (stream, continuation) = AsyncStream<StreamElement>.makeStream()
+        let bufferAllocator = ByteBufferAllocator()
+        self.connectionState = .initial(
+            eventLoopGroup: eventLoopGroup,
+            bufferAllocator: bufferAllocator,
+            requestStream: stream,
+            requestContinuation: continuation
+        )
+    }
+
+    /// Start consuming the requestStream
+    ///
+    /// This function starts consuming the requests from the `requestStream`,
+    /// sending each request to the Memcache server and handling the server's responses.
+    public func run() async throws {
+        guard case .initial(let eventLoopGroup, let bufferAllocator, let stream, let continuation) = connectionState else {
+            return
```

Let's throw an error here